src/engine/source/builder/src/builders/opmap/opBuilderHelperMap.cpp

```diff
@@ -550,13 +550,19 @@ MapBuilder opBuilderHelperStringConcat(bool atleastOne)
                 if (buildCtx->validator().hasField(ref->dotPath()))
                 {
                     auto jtype = buildCtx->validator().getJsonType(ref->dotPath());
-                    if (jtype != json::Json::Type::String)
-                    {
-                        throw std::runtime_error(
-                            fmt::format("Expected 'string' reference but got reference '{}' of type '{}'",
-                                        ref->dotPath(),
-                                        json::Json::typeToStr(jtype)));
-                    }
+                       switch (jtype)
```

Comment:
nitpick: It doesn't seem necessary to use switch here. Isn't an if-else clearer?
```
if (jtype != json::Json::Type::String && 
    jtype != json::Json::Type::Number && 
    jtype != json::Json::Type::Object) 
{
    throw std::runtime_error(fmt::format(
        "Expected 'string/number/object' reference but got reference '{}' of type '{}'",
        ref->dotPath(),
        json::Json::typeToStr(jtype))); // Assuming json::Json::typeToStr is a valid function in the json namespace
}
```